components/Embed/Embed.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * A NextJS Embed Component
+ *
+ * Credit where credit is deserved.
+ * @see: https://github.com/christo-pr/dangerously-set-html-content
+ *
+ * Use this widget with caution. There are no safeguards on what it can do. It
+ * is also not good practice to inject and manipulate the page outside of
+ * REACT as that can lead to irregularities and troubles.
+ */
+import React, { useRef, useEffect } from 'react';
+import { WidthBox, type WidthType } from '../WidthBox';
+import { type PaddingType } from '@/utilities/datasource';
+
+export interface EmbedProps extends  React.HTMLAttributes<HTMLDivElement> {
+  id?: string;
+  src?: string;
+  content?: string;
+  boundingWidth?: 'site' | 'full';
+  width?: WidthType;
+  spacingTop?: PaddingType;
+  spacingBottom?: PaddingType;
+  className?: string;
+}
+
+const Embed = ({
+id, src, content, boundingWidth, spacingTop, spacingBottom, className, width, ...props
+}:EmbedProps) => {
+
+  const myEmbed = useRef(null);
+
+  useEffect(() => {
```

Comment:
I went with the script element and dom fragment approach as it gave me control over putting these scripts inline with the HTML whereas the next script component did not.

Reply:
Was just about to ask about the Next Script - guess the old approach is still better huh 👀